python/python/bystro/covariance/_base_covariance.py

```diff
@@ -0,0 +1,549 @@
+from datetime import datetime as dt
+import pytz
+
+class BaseCovariance(object):
+    """
+    This object basically contains all the methods asides for fitting
+    a covariance matrix.
+    """
+
+    def __init__(self):
+        self.creationDate = dt.now(pytz.timezone("US/Pacific"))
+
+    #################
+    # Miscellaneous #
+    #################
+    def get_precision(self):
+        return _get_precision(self.covariance)
+
+    def get_stable_rank(self):
+        return _get_stable_rank(self.covariance)
+
+    def predict(self, Xobs, idxs):
+        return _predict(self.covariance, Xobs, idxs)
+
+    #####################################
+    # Gaussian Likelihood-based methods #
+    #####################################
+    def conditional_score(self, X, idxs, weights=None):
+        return _conditional_score(self.covariance, X, idxs, weights=weights)
+
+    def conditional_score_samples(self, X, idxs):
+        return _conditional_score_samples(self.covariance, X, idxs)
+
+    def marginal_score(self, X, idxs, weights=None):
+        return _marginal_score(self.covariance, X, idxs, weights=weights)
+
+    def marginal_score_samples(self, X, idxs):
+        return _marginal_score_samples(self.covariance, X, idxs)
+
+    def score(self, X, weights=None):
+        return _score(self.covariance, X, weights=weights)
+
+    def score_samples(self, X):
+        return _score_samples(self.covariance, X)
+
+    #################################
+    # Information-theoretic methods #
+    #################################
+    def entropy(self):
+        return _entropy(self.covariance)
+
+    def entropy_subset(self, idxs):
+        return _entropy_subset(self.covariance, idxs)
+
+    def mutual_information(self, idxs1, idxs2):
+        return _mutual_information(self.covariance, idxs1, idxs2)
+
+
+###########################################
+###########################################
+###########################################
+###                                     ###
+###                                     ###
+###   Methods for covariance matrices   ###
+###                                     ###
+###                                     ###
+###########################################
+###########################################
+###########################################
+
+
+def _get_precision(covariance):
+    """
+    Gets the precision matrix defined as the inverse of the covariance
+
+    Parameters
+    ----------
+    covariance : np.array-like(p,p)
+        The covariance matrix
+
+    Returns
+    -------
+    precision : np.array-like(sum(p),sum(p))
+        The inverse of the covariance matrix
+    """
+    precision = la.inv(covariance)
+    return precision
+
+
+def _get_stable_rank(covariance):
+    """
+    Returns the stable rank defined as
+    ||A||_F^2/||A||^2
+
+    Parameters
+    ----------
+    covariance : np.array-like(p,p)
+        The covariance matrix
+
+    Returns
+    -------
+    srank : float
```

Comment:
nit: is this abbreviation an established convention in numerical linear algebra?  my own preference would be to spell it out for the benefit of reading speed.

Reply:
Done.

Reply:
Appreciate it, but have you pushed the latest commit?  I'm still seeing `srank`on my end